File: mobile/lib/features/gyms/data/gym_photo.dart
class GymPhoto {
  final String id;
  final String url;
  final int sortOrder;
  final String? altTextEn;
  final String? altTextAr;

  const GymPhoto({
    required this.id,
    required this.url,
    required this.sortOrder,
    this.altTextEn,
    this.altTextAr,
  });

  factory GymPhoto.fromJson(Map<String, dynamic> json) {
    return GymPhoto(
      id: json['id'] as String,
      url: json['url'] as String,
      sortOrder: (json['sortOrder'] ?? json['sort_order'] ?? 0) as int,
      altTextEn: json['altTextEn'] as String? ?? json['alt_text_en'] as String?,
      altTextAr: json['altTextAr'] as String? ?? json['alt_text_ar'] as String?,
    );
  }
}

File: backend/app/db/models/gym_photo.py
from __future__ import annotations

from sqlalchemy import ForeignKey, Index, text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.db.types import TimestampTZ, TimestampTZUpdate, UUIDCol, UUIDFk


class GymPhoto(Base):
    __tablename__ = "gym_photos"

    id: Mapped[UUIDCol]
    gym_id: Mapped[UUIDFk] = mapped_column(
        ForeignKey("gyms.id", ondelete="CASCADE"), nullable=False
    )
    url: Mapped[str] = mapped_column(nullable=False)
    sort_order: Mapped[int] = mapped_column(
        nullable=False, default=0, server_default=text("0")
    )
    alt_text_en: Mapped[str | None] = mapped_column(nullable=True)
    alt_text_ar: Mapped[str | None] = mapped_column(nullable=True)

    created_at: Mapped[TimestampTZ]
    updated_at: Mapped[TimestampTZUpdate]

    __table_args__ = (
        Index("ix_gym_photos_gym_sort", "gym_id", "sort_order"),
    )
